scheduler/feasible_test.go

```diff
@@ -390,88 +396,104 @@ func TestHostVolumeChecker_Sticky(t *testing.T) {
 		},
 	}
 
-	dhv := &structs.HostVolume{
+	dhv1 := &structs.HostVolume{
 		Namespace:             structs.DefaultNamespace,
 		ID:                    uuid.Generate(),
 		Name:                  "foo",
 		NodeID:                nodes[1].ID,
 		RequestedCapabilities: hostVolCapsReadWrite,
 		State:                 structs.HostVolumeStateReady,
 	}
+	dhv2 := &structs.HostVolume{
+		Namespace:             structs.DefaultNamespace,
+		ID:                    uuid.Generate(),
+		Name:                  "foobar",
+		NodeID:                nodes[1].ID,
+		RequestedCapabilities: hostVolCapsReadWrite,
+		State:                 structs.HostVolumeStateReady,
+	}
 
 	nodes[0].HostVolumes = map[string]*structs.ClientHostVolumeConfig{}
-	nodes[1].HostVolumes = map[string]*structs.ClientHostVolumeConfig{"foo": {ID: dhv.ID}}
+	nodes[1].HostVolumes = map[string]*structs.ClientHostVolumeConfig{
+		"foo":    {ID: dhv1.ID},
+		"foobar": {ID: dhv2.ID},
+	}
 
 	for _, node := range nodes {
 		must.NoError(t, store.UpsertNode(structs.MsgTypeTestSetup, 1000, node))
 	}
-	must.NoError(t, store.UpsertHostVolume(1000, dhv))
+	must.NoError(t, store.UpsertHostVolume(1000, dhv1))
 
-	stickyRequest := map[string]*structs.VolumeRequest{
+	stickyRequests := map[string]*structs.VolumeRequest{
 		"foo": {
 			Type:           "host",
 			Source:         "foo",
 			Sticky:         true,
 			AccessMode:     structs.CSIVolumeAccessModeSingleNodeWriter,
 			AttachmentMode: structs.CSIVolumeAttachmentModeFilesystem,
 		},
+		"foobar": {
+			Type:           "host",
+			Source:         "foobar",
+			Sticky:         true,
+			AccessMode:     structs.CSIVolumeAccessModeSingleNodeWriter,
+			AttachmentMode: structs.CSIVolumeAttachmentModeFilesystem,
+		},
```

Comment:
I don't think this is the test we're looking for: this means both allocs want 2 volumes. We're more interested in the case where allocs want a single volume each but we want to make sure we're fulfilling existing claims before picking up new ones.

Reply:
fixed in 34f139e